external/avro/src/main/scala/org/apache/spark/sql/avro/package.scala

```diff
@@ -31,10 +32,32 @@ package object avro {
    * @since 2.4.0
    */
   @Experimental
-  def from_avro(data: Column, jsonFormatSchema: String): Column = {
-    new Column(AvroDataToCatalyst(data.expr, jsonFormatSchema))
+  def from_avro(
+      data: Column,
+      jsonFormatSchema: String): Column = {
+    new Column(AvroDataToCatalyst(data.expr, jsonFormatSchema, Map.empty))
+  }
+
+  /**
+   * Converts a binary column of avro format into its corresponding catalyst value. The specified
+   * schema must match the read data, otherwise the behavior is undefined: it may fail or return
+   * arbitrary result.
+   *
+   * @param data the binary column.
+   * @param jsonFormatSchema the avro schema in JSON string format.
+   * @param options options to control how the Avro record is parsed.
+   *
+   * @since 3.0.0
+   */
+  @Experimental
+  def from_avro(
+      data: Column,
+      jsonFormatSchema: String,
+      options: Map[String, String]): Column = {
```

Comment:
One thing I am less sure is tho, what do you think about we just expose one Java specific one? This will only be usable in Scala side.

Reply:
ah that's a good point, as that's what we do for other APIs as well.

We can think for about this later, and change all the APIs together, if other people complains.

Reply:
+1 with @cloud-fan